### DISCUSSION TITLE
What exactly do I need a license for?

### DISCUSSION THREAD

**Discussion by nexensys** [remotion-dev/remotion#1923] [Q&A] [answered]
I see from the website that temptation is free for individuals, and you need a license if you're a company. Cool. Great for solo devs. However, what if I wanted to use remotion as a central component of a project (a free, open source, extensible video editor in this case) would I need a license? No profit would be made a side from maybe donations through GitHub sponsors.

**Comment by JonnyBurger** [comment, marked as answer]
Hi @ErrorGamer2000!

You can use the free license for teams up to three people if it is a personal project even if you still work at a company.

If the project is part of a company, we require you to buy a license, because usually companies have commercial interest in everything that they do, even their open source projects. We have exceptions for non-profits though.

**Reply by nexensys** [reply]
See, this is kinda a grey area. I'm not a team, and I'm not doing this as part of a company. This is a project I will be working on by myself, but would like to share with others via open source. So, anybody could use it, from individuals, to teams, to companies. Would it work if I put a link in a "before you use this" section and tell them to make sure they have the correct license?

**Reply by JonnyBurger** [reply]
Got it! 

Yes you can totally make open source projects using Remotion - preferrably you add a license disclaimer similar to this one: https://github.com/remotion-dev/github-unwrapped-2022#license

**Reply by nexensys** [reply]
Sweet! Thank you!

**Reply by nexensys** [reply]
@JonnyBurger what if I decided to make said editor closed source and make the app paid? They will still need a license from remotion, but will also have to pay to use the app.

**Reply by JonnyBurger** [reply]
This would be another case - people would not install a Remotion project anymore, they would install your app which uses Remotion as a dependency.

You can totally do this with the free license! As long as people don't directly edit the React code, this is fine.

**Reply by nexensys** [reply]
Well, that's a little bit of a grey area... I want to allow people to make their own effects using react, and be able to put them on a sort of public store...

**Reply by JonnyBurger** [reply]
@ErrorGamer2000 I see, if they actually write their own remotion code, then it is a bit special.. then companies need a license! If they create effects in other ways (they don't import remotion) them I think free license is fine.

The basic idea is to be as permissive as possible, but prevent people making derivates of Remotion or mask the awareness for others that they are writing remotion code and would need a license.

**Reply by nexensys** [reply]
@JonnyBurger How about instead of them importing remotion hooks I add a sort of api wrapper, keeping the remotion logic hidden?

**Reply by JonnyBurger** [reply]
I agree there is some gray area, thanks for asking for clarification – I say you can do wrapping and that is fine! It seems like a cool project and we don't want to hold you back with license issues.

**Reply by nexensys** [reply]
Thanks @JonnyBurger

**Reply by nexensys** [reply]
Also, @JonnyBurger I would very much like to be able to credit you/remotion. What would you like me to add and where would you like me to add it? Or would it only cause more license issues?

**Reply by JonnyBurger** [reply]
That would be awesome if you mention us of course! 🙌

**Reply by nexensys** [reply]
Does it matter where, or can I just put it in the developer credits on the website (since it is going to be closed-source, if free)

**Reply by JonnyBurger** [reply]
@ErrorGamer2000 We have no guidelines on where to put it, so do as you please!
